core/services/job/orm.go

```diff
@@ -915,11 +971,7 @@ WHERE id = $1
 // CountPipelineRunsByJobID returns the total number of pipeline runs for a job.
 func (o *orm) CountPipelineRunsByJobID(jobID int32) (count int32, err error) {
 	err = o.q.Transaction(func(tx pg.Queryer) error {
-		stmt := `
-SELECT COUNT(*)
-FROM pipeline_runs
-WHERE pipeline_runs.pipeline_spec_id = (SELECT jobs.pipeline_spec_id FROM JOBS WHERE jobs.id = $1)
-`
+		stmt := "SELECT COUNT(*) FROM pipeline_runs JOIN jobs USING (pipeline_spec_id) WHERE jobs.id = $1"
```

Comment:
Note this is still linear and slow

Reply:
With the BRIN index, this was already noticeably faster than the other query since it was O(N) while the other one was O(N log N) due to the Sort operation.  Neither of them was able to make use of the index (for different reasons I think).

Switching to BTREE allowed this query to do an Index-only scan which sped it up a lot.  I'm not sure how that works, so I don't know if it's still technically linear or not, I just know that it's very fast now.

Switching to BTREE also allowed the other query to use the index, but after that it was still pretty slow... in fact, much slower than the COUNT query.  Just using the index was enough to make this query fast, but not the other query.  Which is why the more complex work on the other query was necessary (breaking it into batches).  Now they are both nice and fast.